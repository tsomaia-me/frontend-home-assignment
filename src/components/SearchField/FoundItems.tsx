import React from 'react'
import { observer } from 'mobx-react'
import { usePackageStore } from '../../hooks/usePackageStore'
import styles from './SearchField.module.css'
import FoundItem from './FoundItem'

export interface FoundItemsProps {
  onSelect: (packageName: string) => void
}

function FoundItems({ onSelect }: FoundItemsProps) {
  const packageStore = usePackageStore()

  return (
    <ul className={styles.list}>
      {packageStore.foundPackages.map(packageName => (
        <FoundItem
          key={packageName}
          packageName={packageName}
          onClick={onSelect}
        />
      ))}
    </ul>
  )
}

export default observer(FoundItems)
