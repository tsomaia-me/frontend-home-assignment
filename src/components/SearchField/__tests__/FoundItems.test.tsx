import React from 'react'
import renderer from 'react-test-renderer'
import FoundItems from '../FoundItems'

test('renders FoundItems correctly', () => {
  const json = renderer.create(<FoundItems onSelect={console.log}/>).toJSON()
  expect(json).toMatchSnapshot()
})
