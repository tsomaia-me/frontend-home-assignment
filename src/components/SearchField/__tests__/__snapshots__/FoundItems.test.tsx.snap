// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders FoundItems correctly 1`] = `
<ul
  className="list"
/>
`;
